### PR TITLE
Add MkDocs Material developer documentation (#695)

### PR DESCRIPTION
**Pull request by YuanYuan98**
Set up MkDocs with the Material theme for developer documentation. The docs source lives in `docs-src/`, builds to `docs/api/`, and is hosted alongside the existing Samudra 2 paper page on GitHub Pages.

  ## Changes
  - `mkdocs.yml` — config with Material theme, mkdocstrings for auto-generated API reference
  - `docs-src/` — Markdown source (Getting Started, Architecture, API Reference for all models including Samudra, Samudra 2, and FOMO)
  - `.github/workflows/pages.yml` — updated to build MkDocs on deploy
  - `docs/index.html` — added API Docs card on landing page
  - `pyproject.toml` — added `docs` dependency group (mkdocs, mkdocs-material, mkdocstrings)

  ## Preview locally
  ```bash
  uv sync --group docs
  uv run mkdocs serve

  Note

  Build artifacts (docs/api/) are temporarily committed so reviewers can preview without building locally. Once CI is stable, these can be moved to .gitignore and built only in CI.

  Built with the help of Claude Code.
  ```

### CODE REVIEW

**Comment by alxmrs** [comment]
High level feedback: It seems like [mkdocs 1 is unmaintained](https://github.com/mkdocs/mkdocs/discussions/4010), and [mkdocs 2 is not battle tested](https://github.com/encode/mkdocs). When I compile the documentation, it links to [this blog post](https://squidfunk.github.io/mkdocs-material/blog/2026/02/18/mkdocs-2.0/), which says that [zensicle is a drop-in replacement](https://squidfunk.github.io/mkdocs-material/blog/2026/02/18/mkdocs-2.0/#where-zensical-comes-in) for mkdocs 1. 

In light of this, I think we should adopt zensicle as our doc generator.

**Comment by alxmrs** [comment]
🐑  Optional point of style feedback (you may ignore this): I am not a fan of material design.

File: docs-src/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+**A PyTorch-based machine learning framework for training and evaluating neural ocean emulators.**
+
+Ocean Emulators provides tools to build models that learn to predict future ocean states autoregressively, achieving orders-of-magnitude speedups over traditional ocean general circulation models.
+
+## Models
+
+- **Samudra** — ConvNeXt U-Net architecture for single-scale ocean emulation at 1° resolution
+- **Samudra 2** — Wider ConvNeXt U-Net with dynamic variance-weighted loss, scaling to 1°, 1/2°, and 1/4° resolution with stable ~8-year rollouts
+- **FOMO** — Foundation model with encoder-processor-decoder architecture supporting multi-scale training
```
Comment (alxmrs) [inline docs-src/index.md]:
```suggestion
- **FOMO** — A "Foundation Ocean Model + Observations" uses a encoder-processor-decoder architecture to support multi-scale training
```

File: docs-src/index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Ocean Emulators
+
+**A PyTorch-based machine learning framework for training and evaluating neural ocean emulators.**
+
+Ocean Emulators provides tools to build models that learn to predict future ocean states autoregressively, achieving orders-of-magnitude speedups over traditional ocean general circulation models.
+
+## Models
+
+- **Samudra** — ConvNeXt U-Net architecture for single-scale ocean emulation at 1° resolution
```
Comment (alxmrs) [inline docs-src/index.md]: All of the items in this list (and the list in the next section) should end with a full stop (i.e. a period "`.`").

File: docs-src/getting-started/installation.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+source .venv/bin/activate
+```
+
+## Verify Installation
+
+Run the test suite to confirm everything is set up correctly:
+
+```bash
+uv run pytest -m "not manual and not cuda" -n auto
+```
```
Comment (alxmrs) [inline docs-src/getting-started/installation.md]: Instead of nudging users to run tests (which is more of a concern for development rather than use), let's have user print the help docs for the train CLI: 

```
uv run -m ocean_emulators.train --help
```

File: docs-src/getting-started/quick-start.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Quick Start
```
Comment (alxmrs) [inline docs-src/getting-started/quick-start.md]: This looks like a good _development_ quickstart guide. However, I think readers would rather learn how to get starting running model training/eval/viz or fine tuning. 

Ideally, this quick start would cover a realistic walkthrough of the project. This will probably involve data paths that I haven't made public yet (I'm working on this today). For now, I think it's OK to point users to internal OSN pod data locations (I'll update the paths to clean versions later on).

File: docs-src/concepts/architecture.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# Architecture
+
+## Overview
+
+Ocean Emulators is organized around a few core components that work together to train and evaluate neural ocean emulators.
+
+```
+Input (two ocean states + forcing) → Model → Output (next two predicted states)
```
Comment (alxmrs) [inline docs-src/concepts/architecture.md]: Let's have claude produce an ASCII diagram for our training process.

File: docs-src/concepts/architecture.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+The emulator autoregressively predicts future ocean states. During training, short rollouts (K=4 steps) are used. During inference, the model runs freely for hundreds of steps without ground-truth feedback.
+
+## Core Components
+
+### Models
+
+All models inherit from a common base class (`ocean_emulators.models.base`) that provides:
```
Comment (alxmrs) [inline docs-src/concepts/architecture.md]:
```suggestion
All models inherit from a common base class (`ocean_emulators.models.base`) that provides configuration for:
```

File: docs-src/concepts/architecture.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+Ocean Emulators is organized around a few core components that work together to train and evaluate neural ocean emulators.
+
+```
+Input (two ocean states + forcing) → Model → Output (next two predicted states)
+```
+
+The emulator autoregressively predicts future ocean states. During training, short rollouts (K=4 steps) are used. During inference, the model runs freely for hundreds of steps without ground-truth feedback.
+
+## Core Components
```
Comment (alxmrs) [inline docs-src/concepts/architecture.md]: Ideally, Claude could make an ascii diagram of all the components listed in the subsections below to illustrate how they are interrelated. It may be worthwhile to create our own SVG diagram of the components in Google Drive and then embed it here.

File: docs-src/concepts/architecture.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+- `train_batch` — single training step with loss computation
+- `validate_batch` — validation without gradient updates
+- `inference` — long autoregressive rollouts
+
+### Data Pipeline
+
+`ocean_emulators.datasets` handles OM4 ocean model data in Zarr format:
+
+- `TrainData` — training dataset with time-based splits
+- `InferenceDataset` — evaluation dataset for long rollouts
```
Comment (alxmrs) [inline docs-src/concepts/architecture.md]:
```suggestion
- `InferenceDataset` — evaluation dataset for long rollouts. Only supports a single scale of data.
```

File: docs-src/concepts/architecture.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+- `train_batch` — single training step with loss computation
+- `validate_batch` — validation without gradient updates
+- `inference` — long autoregressive rollouts
+
+### Data Pipeline
+
+`ocean_emulators.datasets` handles OM4 ocean model data in Zarr format:
+
+- `TrainData` — training dataset with time-based splits
```
Comment (alxmrs) [inline docs-src/concepts/architecture.md]:
```suggestion
- `TrainData` — training dataset with time-based splits. Supports single or multiscale training.
```

File: docs-src/concepts/architecture.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+- `InferenceDataset` — evaluation dataset for long rollouts
+- Supports 1°, 1/2°, and 1/4° resolutions
+
+### Configuration
+
+YAML-based configuration with Pydantic validation (`ocean_emulators.config`). Supports `!include` directives and command-line overrides.
```
Comment (alxmrs) [inline docs-src/concepts/architecture.md]: Let's put a link to the section in the contributing guide that explains how to use the configuration system.

File: docs-src/concepts/architecture.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+- Distributed training via PyTorch DDP
+- Checkpointing
+- Learning rate scheduling with warmup
+- EMA (Exponential Moving Average) support
+- Weights & Biases integration
+
+### Evaluation
+
+The aggregator system (`ocean_emulators.aggregator`) collects metrics during training and evaluation, including map metrics, reduced metrics, and snapshot visualizations.
```
Comment (alxmrs) [inline docs-src/concepts/architecture.md]: To mirror one section above, let's discuss what `ocean_emulators.train` does. 

The text for this section doesn't match the title in my opinion; I think the aggregator, if we include it, should be it's own section.

File: docs/models/base.md [hunk 12]
```diff
@@ -0,0 +1,3 @@
+# Base Model
+
+::: ocean_emulators.models.base
```
Comment (alxmrs) [inline docs/models/base.md:3]: I'm surprised that the generated docs don't produce the class information for the base model. Can you look into why that is?

Comment (alxmrs) [reply docs/models/base.md:3]: Is this supposed to automatically pull in the class contract?

File: .github/workflows/pages.yml [hunk 13]
```diff
@@ -1,11 +1,13 @@
-name: Deploy Samudra 2 paper site
+name: Deploy docs site
```
Comment (alxmrs) [inline .github/workflows/pages.yml]: Let's make this something simple like "docs" so we can include a status badge on our README.

File: .github/workflows/pages.yml [hunk 14]
```diff
@@ -1,11 +1,13 @@
...
 
 on:
   push:
     branches:
       - main
     paths:
       - "docs/**"
+      - "docs-src/**"
```
Comment (alxmrs) [inline .github/workflows/pages.yml]: IMO, I think all these contents should live under the `docs/` folder. The previous webpages could be moved to a `static/` folder and we could introduce a CI step to move all the static contents to the project root.

File: docs/api/assets/javascripts/lunr/min/lunr.ar.min.js [hunk 15]
```diff
@@ -0,0 +1 @@
...يت ذينك ذين ذه ذي راح رجع رويدك ريث رب زيارة سبحان سرعان سنة سنوات سوف سوى ساء ساءما شبه شخصا شرع شتان صار صباح صفر صه صه ضد ضمن طاق طالما طفق طق ظل عاد عام عاما عامة عدا عدة عدد عدم عسى عشر عشرة علق على عليك عليه عليها عل عن عند عندما عوض عين عدس عما غدا غير  ف فان فلان فو فى في فيم فيما فيه فيها قال قام قبل قد قط قلما قوة كانما كاين كاي كاين كاد كان كانت كذا كذلك كرب كل كلا كلاهما كلتا كلم كليكما كليهما كلما كلا كم كما كي كيت كيف كيفما كان كخ لئن لا لات لاسيما لدن لدى لعمر لقاء لك لكم لكما لكن لكنما لكي لكيلا للامم لم لما لما لن لنا له لها لو لوكالة لولا لوما لي لست لست لستم لستما لستن لست لسن لعل لكن ليت ليس ليسا ليستا ليست ليسوا لسنا ما ماانفك مابرح مادام ماذا مازال مافتئ مايو متى مثل مذ مساء مع معاذ مقابل مكانكم مكانكما مكانكن مكانك مليار مليون مما ممن من منذ منها مه مهما من من نحن نحو نعم نفس نفسه نهاية نخ نعما نعم ها هاؤم هاك هاهنا هب هذا هذه هكذا هل هلم هلا هم هما هن هنا هناك هنالك هو هي هيا هيت هيا هؤلاء هاتان هاتين هاته هاتي هج هذا هذان هذين هذه هذي هيهات وا واحد واضاف واضافت واكد وان واها واوضح وراءك وفي وقال وقالت وقد وقف وكان وكانت ولا ولم ومن وهو وهي ويكان وي وشكان يكون يمكن يوم ايان".split(" ")),e.Pipeline.registerFunction(e.ar.stopWordFilter,"stopWordFilter-ar")}});
```
Comment (alxmrs) [inline docs/api/assets/javascripts/lunr/min/lunr.ar.min.js]: I hear your reasoning for checking these in, but I am happy to build the files locally. Having the generated files makes this PR a bit difficult to review (not to mention, the commit size much larger than it needs to be). Please remove these and update the .gitignore.

File: docs/index.html [hunk 16]
```diff
@@ -37,7 +37,18 @@ <h2 class="title is-4">Samudra 2</h2>
...
         </div>
-        <div class="column is-half">
+        <div class="column is-one-third">
+          <div class="box feature-card">
+            <h2 class="title is-4">API Docs</h2>
+            <p>
+              Developer documentation and API reference.
+            </p>
+            <p>
+              <a class="button is-dark is-rounded" href="./api/index.html">Open page</a>
```
Comment (alxmrs) [inline docs/index.html]: I prefer the slug `/docs/` over `/api/`. Furthermore, will you omit the "index.html"? That should make a cleaner URL.

File: mkdocs.yml [hunk 17]
```diff
@@ -0,0 +1,88 @@
+site_name: Ocean Emulators
+site_url: https://open-athena.github.io/Ocean_Emulator/api/
```
Comment (alxmrs) [inline mkdocs.yml]: ideally, the URL would be: 
```suggestion
site_url: https://open-athena.github.io/Ocean_Emulator/docs/
```

File: mkdocs.yml [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+            docstring_style: google
+
+nav:
+  - Home: index.md
+  - Getting Started:
+    - Installation: getting-started/installation.md
+    - Quick Start: getting-started/quick-start.md
+  - Concepts:
+    - Architecture: concepts/architecture.md
+  - API Reference:
```
Comment (alxmrs) [inline mkdocs.yml:44]: Please include the contributing guide in the documentation.

File: docs/models/modules.md [hunk 19]
```diff
@@ -0,0 +1,21 @@
+# Modules
```
Comment (alxmrs) [inline docs/models/modules.md:1]: These generated docs look pretty good!

File: mkdocs.yml [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+    - Models:
+      - Base Model: models/base.md
+      - Samudra: models/samudra.md
+      - FOMO: models/fomo.md
+      - Modules: models/modules.md
+    - Data:
+      - Datasets: datasets.md
```
Comment (alxmrs) [inline mkdocs.yml]: Since this is just one item, lets omit the nesting. If we need to expand the data category, we can do it later.

File: docs-src/config.md [hunk 21]
```diff
@@ -0,0 +1,5 @@
+# Configuration
```
Comment (alxmrs) [inline docs-src/config.md]: I think we should move the text on how to use configuration from the contributing guide and put it on the top of this file. I agree that we should have the API reference classes here, too.

File: docs/eval.md [hunk 22]
```diff
@@ -0,0 +1,3 @@
+# Eval
+
+::: ocean_emulators.eval
```
Comment (alxmrs) [inline docs/eval.md:3]: I'm not sure why the mkdocs didn't parse this.

File: docs/utils/loss.md [hunk 23]
```diff
@@ -0,0 +1,28 @@
...
+# Dynamic variance-weighted loss (Samudra 2)
+loss:
+  type: dynamic
+  metric: mse
+  limit: 20
+```
+
+## API Reference
+
+::: ocean_emulators.utils.loss
```
Comment (alxmrs) [inline docs/utils/loss.md:28]: I find the template we're using to produce really confusing generated documentation for classes. For example, here is what the dynamic loss class looks. I find it really hard to understand if the properties here are actually part of the class (knowing the codebase, they are). 

Image

**Review by alxmrs — CHANGES_REQUESTED** [review]
Here are some high level feedback from rendering the docs locally. 

One thing we should consider doing in addition to the comments I've made here is extracting core parts of the CONTRIBUTING guide into their own documentation pages.

File: .github/workflows/pages.yml [hunk 24]
```diff
@@ -24,13 +26,22 @@ jobs:
...
         uses: actions/checkout@v5
 
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+
+      - name: Build API docs with Zensical
+        run: uv run --only-group docs zensical build
+
+      - name: Copy static pages into build output
+        run: cp -r static/* site/
```
Comment (alxmrs) [inline .github/workflows/pages.yml]: Let's store static inside docs because they are documentation-specific.
```suggestion
        run: cp -r docs/static/* site/
```

File: mkdocs.yml [hunk 25]
```diff
@@ -0,0 +1,78 @@
...
+    - Utilities:
+      - Data Utils: utils/data.md
+      - Distributed: utils/distributed.md
+      - Loss: utils/loss.md
+      - EMA: utils/ema.md
+      - Schedule: utils/schedule.md
+      - Samplers: utils/samplers.md
+      - Writer: utils/writer.md
+    - Visualization: viz.md
+  - Contributing: contributing.md
```
Comment (alxmrs) [inline mkdocs.yml]: Let's put this before API reference; the way it is now it looks like this guide is part of the API.

**Review by alxmrs — APPROVED** [review]
Things to do before merge: 
- Move the static/ folder into docs/ (and fix the pages deploy)
- Address the current failing checks

Otherwise, this is an excellent documentation website! Thank you for building and writing it.

File: .github/workflows/pages.yml [hunk 26]
```diff
@@ -24,13 +25,22 @@ jobs:
...
         uses: actions/checkout@v5
 
+      - name: Install uv
+        uses: astral-sh/setup-uv@v6
+
+      - name: Build API docs with Zensical
+        run: uv run --only-group docs zensical build
+
+      - name: Copy static pages into build output
+        run: cp -r docs/static/* site/
```
Comment (alxmrs) [inline .github/workflows/pages.yml:35]: Thanks! Perfect.
